testsuite/classlibrary/TestSoundFile.sc

```diff
@@ -53,4 +53,40 @@ TestSoundFile : UnitTest {
 
 	}
 
+	test_cue {
+
+		var cueEvent, testEvent, group;
+		var server = Server(this.class.name);
+		this.bootServer(server);
+
+		// pass in Event
+		testEvent = (foo: true);
+		cueEvent = soundFile.cue(testEvent);
+		this.assert(cueEvent[\foo], "Cue should take an Event as first argument");
+
+		// playNow argument
+		cueEvent = soundFile.cue((), true);
+		0.5.wait;
```

Comment:
is there any way to wait on something specific here (other than 0.5)? if not, we should add a callback argument to `cue`.

Reply:
I'm sure there's a way to avoid waiting. I can look into this test again next week to see if I can improve it.

Reply:
Formally, the right thing to wait for would be `/n_go` coming back from the server. But, if you're waiting for that, then the test needs to fail if `/n_go` doesn't come back. The only way I can think of is to include a timer.

If you don't handle the case of a broken server not sending `/n_go` back (or any other reply), then the entire test suite hangs at this point. You can use the incoming reply to move forward with the test sooner than the wait time, but you can't get rid of a wait time altogether.

Reply:
> Formally, the right thing to wait for would be /n_go coming back from the server. But, if you're waiting for that, then the test needs to fail if /n_go doesn't come back. The only way I can think of is to include a timer.

Then let's do that please. In the last few months we've established a pattern in tests for dealing with callback timeout - `fork` a timeout thread that signals a condition variable. Since there is so so much async callback code in SC I think we should probably find a way to simplify it.